typings/react-popper.d.ts

```diff
@@ -6,28 +6,36 @@ interface ManagerProps {
 }
 export class Manager extends React.Component<ManagerProps, {}> { }
 
+type RefHandler = (ref: HTMLElement | null) => void;
+
+interface ReferenceChildrenProps {
+  ref: RefHandler;
+}
+
 interface ReferenceProps {
-  children: (props: ({
-    ref: (ref: HTMLElement | null) => void,
-  })) => React.ReactNode;
+  children: (props: ReferenceChildrenProps) => React.ReactNode;
 }
 export class Reference extends React.Component<ReferenceProps, {}> { }
 
+interface PopperArrowProps {
+  ref: RefHandler;
+  style: React.CSSProperties;
+}
+
+interface PopperChildrenProps {
+  arrowProps: PopperArrowProps;
+  outOfBoundaries: boolean | null;
+  placement?: PopperJS.Placement;
```

Comment:
@FezVrasta do we know if this field will always be defined? as in, should it be marked optional or not? my suspicion is that it's always defined, that popper.js always computes a valid `placement` regardless of the given option value

Reply:
Yes actually I think it's always going to be defined, I'm not sure why I marked it as optional.